internal/controller/linodefirewall_controller_helpers.go

```diff
@@ -162,34 +234,62 @@ func transformToCIDR(ip string) string {
 }
 
 // processInboundRule handles a single inbound rule
-func processInboundRule(rule infrav1alpha2.FirewallRule, createOpts *linodego.FirewallCreateOptions) {
-	ruleIPv4s, ruleIPv6s := processAddresses(rule.Addresses)
+func processInboundRule(ctx context.Context, k8sClient clients.K8sClient, log logr.Logger, rule infrav1alpha2.FirewallRule, lfw *infrav1alpha2.LinodeFirewall, createOpts *linodego.FirewallCreateOptions) error {
+	var ruleIPv4s []string
+	var ruleIPv6s []string
+	var err error
+	if rule.Addresses != nil {
+		ruleIPv4s, ruleIPv6s = processAddresses(rule.Addresses)
+	} else if rule.AddressSetRefs != nil {
```

Comment:
should this be elseIf ? if the intent is to support both?

Reply:
looks like that was leftover, outbound was updated but not inbound